### ISSUE TITLE
Unable to run the result of generate_installer.sh due to undefined variable 

### ISSUE THREAD

**Issue by johntdyer** [CGamesPlay/infra#2]
When I run generate_script.sh I get the following:

```bash
13:42:42 jdyer@coruscant.local infra master ? ./bootstrap/generate_installer.sh | sudo bash
++ hostname
+ instance_id=coruscant.local
+ tee /usr/local/share/ca-certificates/global.vault.crt
+ update-ca-certificates
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...

done.
Updating Mono key store
Mono Certificate Store Sync - version 6.8.0.105
Populate Mono certificate store from a concatenated list of certificates.
Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD licensed.

Importing into legacy system store:
I already trust 133, your new list has 129
Import process completed.

Importing into BTLS system store:
I already trust 133, your new list has 129
Import process completed.
Done
done.
+ tee /etc/wireguard/wg0.conf
++ ip -o -4 route show to default
++ awk '{print $5}'
++ head -n 1
+ sed -i s/IFACE_NAME/eno1/g /etc/wireguard/wg0.conf
+ sed -i 's/.*net\.ipv4\.ip_forward.*/net.ipv4.ip_forward=1/' /etc/sysctl.conf
+ sysctl -p
net.ipv4.ip_forward = 1
vm.dirty_background_ratio = 5
vm.dirty_ratio = 10
net.core.rmem_default = 8388608
net.core.rmem_max = 16777216
fs.inotify.max_user_watches = 100000
net.ipv4.ip_forward = 1
net.ipv4.conf.all.forwarding = 1
net.ipv4.igmp_max_memberships = 80
net.ipv4.neigh.default.gc_thresh1 = 8096
net.ipv4.neigh.default.gc_thresh2 = 12288
net.ipv4.neigh.default.gc_thresh3 = 16384
net.ipv4.neigh.default.gc_stale_time = 30
+ ufw allow 51820/udp
Skipping adding existing rule
+ systemctl enable --now wg-quick@wg0
+ wg set wg0 peer Lr/ONxhiViAzDNBlsch9JMPGNpMBc1lB1+ydRk/16VI= allowed-ips 172.30.15.1/32
+ chmod 750 /opt/consul /etc/consul.d
+ tee /opt/consul/agent.crt
+ tee /opt/consul/agent.key
+ echo node_name = '"server-coruscant.local"'
+ echo encrypt = NUJqmicTDl4a+ZjcSg5ih0kN5OEyqBwsym3KtO2ShRE=
encrypt = NUJqmicTDl4a+ZjcSg5ih0kN5OEyqBwsym3KtO2ShRE=
+ tee -a /etc/consul.d/consul.hcl
+ tee /etc/consul.d/server.hcl
+ tee /etc/consul.d/client.hcl
+ tee /etc/systemd/system/consul.service
+ tee -a /etc/systemd/resolved.conf
+ chmod 750 /opt/vault /etc/vault.d
+ rm -rf /opt/vault/agent.crt /opt/vault/agent.key
+ tee /opt/vault/agent.crt
+ tee /opt/vault/agent.key
+ tee /etc/vault.d/vault.hcl
+ tee /etc/systemd/system/vault.service
+ export VAULT_ADDR=https://127.0.0.1:8200
+ VAULT_ADDR=https://127.0.0.1:8200
+ export VAULT_CACERT=/usr/local/share/ca-certificates/global.vault.crt
+ VAULT_CACERT=/usr/local/share/ca-certificates/global.vault.crt
+ sed -i '/^VAULT_ADDR=/d' /etc/environment
+ echo VAULT_ADDR=https://127.0.0.1:8200
+ mkdir --parents /etc/vault-agent.d
+ chmod 750 /etc/vault-agent.d
+ tee /etc/vault-agent.d/agent.crt
+ tee /etc/vault-agent.d/agent.key
+ tee /usr/local/bin/cert-validity-secs
+ chmod +x /usr/local/bin/cert-validity-secs
+ tee /etc/vault-agent.d/rotate-certificates.hcl
bash: line 471: wireguard_ip: unbound variable
```

Clearly wireguard_ip is needed and I see it in generate_installer, but I am just unsure why its not working when folllowing your guide 

Thanks

**Comment by CGamesPlay** [comment]
Hmm, I think I see the problem. Can you verify that this change resolves the issue? Basically, it was trying to do a variable substitution remotely rather than doing it locally.

```diff
diff --git i/bootstrap/generate_installer.sh w/bootstrap/generate_installer.sh
index 10782c7..d7dad45 100755
--- i/bootstrap/generate_installer.sh
+++ w/bootstrap/generate_installer.sh
@@ -347,7 +347,7 @@ template {
 EOF
 }
 END_FILE
-cat <<'EOF'
+cat <<EOF
 sed -i 's/WIREGUARD_IP/'$wireguard_ip'/g' /etc/vault-agent.d/rotate-certificates.hcl

 EOF
```

By the way, the output of `generate_installer.sh` is designed to be manually inspected and run in step. Although it should work when run all at once for automation, I designed it with the use case of manually copying and pasting commands into an ssh session.

Hopefully this fixes it for you, and if so I'll commit it. Let me know if you encounter any other issues!

**Comment by CGamesPlay** [comment]
I'm gonna mark this closed, since I'm pretty sure that was the issue. Thanks for reporting, feel free to open another issue if you have any more troubles!

Closed in bb956b7.
